### PR TITLE
WFLY-9447 NPE in ejb client and clean shutdown 

### PR DESCRIPTION
**Pull request by rhusar**
Jira
https://issues.jboss.org/browse/WFLY-9447

### CODE REVIEW

**Comment by rhusar** [comment]
@pferraro Please approve.

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/DeploymentRepositoryService.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        modules = Collections.emptyMap();
+    }
+
+    @Override
+    public void stop(StopContext context) {
+        modules = Collections.emptyMap();
+    }
+
+    @Override
+    public DeploymentRepository getValue() throws IllegalStateException, IllegalArgumentException {
```
Comment (pferraro) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/DeploymentRepositoryService.java]: You can drop these from the throws clause.  Not only are they not thrown, but they are RuntimeExceptions.

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/DeploymentRepositoryService.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+
+
+/**
+ * Repository for information about deployed modules. This includes information on all the deployed EJB's in the module
+ *
+ * @author Stuart Douglas
+ */
+public class DeploymentRepositoryService implements DeploymentRepository, Service<DeploymentRepository> {
```
Comment (pferraro) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/DeploymentRepositoryService.java]: :+1: Excellent.  WF has too many poorly encapsulated services that expose the Service interface to consumers.

**Review by pferraro — CHANGES_REQUESTED** [review]
One minor nitpick, but otherwise looks good.

**Review by pferraro — APPROVED** [review]

**Comment by tommaso-borgato** [comment]
@rhusar I re-run the affected test and the error does not occur any more

**Comment by rhusar** [comment]
@tommaso-borgato Great, thanks.

**Pull request merged**